bug: Makefile calls for outdated release of logcheck

### Describe the bug

`logcheck -check-contextual` used to complain about uses of `klog.InitFlags`, even though those are not actually violations of contextual logging.  Witness the bogus complaints in `hack/logcheck.out`.  The latest release, v0.4.0, includes a fix to this.

### Steps To Reproduce

1.  Look for complaints about InitFlags in `hack/logcheck.out`

### Expected Behaviour

I expected no objections to using klog.InitFlags.

### Additional Context

The logcheck fix is https://github.com/kubernetes-sigs/logtools/pull/10

I tried switching to the latest release of logcheck, and many of its complaints are about different things.

The first example is the key "resource-types".  I presume that kcp could and should switch to using "resourceTypes" instead, to conform with the Kube norms.

The second example is "VirtualWorkspaceURL".  I presume that cound and should be changed to "virtualWorkspaceURL".

Here is another complaint:

```
/pkg/reconciler/core/logicalclusterdeletion/logicalcluster_deletion_controller.go:316:14: Key positional arguments "owner.cluster" are expected to no special characters. Please refer to https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments.
```

If I understand correctly, that could and should be changed to "ownerCluster".

In fact, I do not see any example where the key is clearly controllerName + "." + controller-specific variable.  Perhaps ones like the following?

```
/pkg/virtual/syncer/transformations/transformer.go:110:11: Key positional arguments "syncerView.HasSyncerFinalizer" are expected to no special characters. Please refer to https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments.
```

Could that be changed to "syncerViewHasSyncerFinalizer" ?

--- a/Makefile
+++ b/Makefile
@@ -57,7 +57,7 @@ GOTESTSUM_VER := v1.8.1
 GOTESTSUM_BIN := gotestsum
 GOTESTSUM := $(abspath $(TOOLS_DIR))/$(GOTESTSUM_BIN)-$(GOTESTSUM_VER)
 
-LOGCHECK_VER := v0.2.0
+LOGCHECK_VER := v0.4.0
 LOGCHECK_BIN := logcheck
 LOGCHECK := $(TOOLS_GOBIN_DIR)/$(LOGCHECK_BIN)-$(LOGCHECK_VER)
 export LOGCHECK # so hack scripts can use it

--- a/hack/verify-contextual-logging.sh
+++ b/hack/verify-contextual-logging.sh
@@ -22,17 +22,18 @@ REPO_ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")"/.. && pwd)
 LOG_FILE="${REPO_ROOT}/hack/logcheck.out"
 work_file="$(mktemp)"
 LOGCHECK=${LOGCHECK:-logcheck}
+LOGCHECK_ARGS="-check-contextual -check-key=false"
 
 cd "$REPO_ROOT"
 
 set +o errexit
-${LOGCHECK} -check-contextual ./... > "${work_file}" 2>&1
+${LOGCHECK} ${LOGCHECK_ARGS} ./... > "${work_file}" 2>&1
 set -o errexit
 
 # pkg/apis is a separate module, so check that in addition to our root packages
 cd "${REPO_ROOT}"/pkg/apis
 set +o errexit
-${LOGCHECK} -check-contextual ./... >> "${work_file}" 2>&1
+${LOGCHECK} ${LOGCHECK_ARGS} ./... >> "${work_file}" 2>&1
 set -o errexit
 
 is_gnu_sed() { sed --version >/dev/null 2>&1; }

